google/cloud/pubsublite/batching_options.h

```diff
@@ -0,0 +1,76 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_BATCHING_OPTIONS_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUBLITE_BATCHING_OPTIONS_H
+
+#include "google/cloud/version.h"
+#include <algorithm>
+#include <chrono>
+
+namespace google {
+namespace cloud {
+namespace pubsublite {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * Batching options for a `Publisher`.
+ */
+class BatchingOptions {
```

Comment:
I think it makes it clearer to group related options, even if they're passed at the top level as a field in google::cloud::Options. Do you disagree? If so, do you have a problem with this struct existing in pubsublite/internal so classes can declare which sets of options they care about, even if at the top level those are parsed from individual google::cloud::Options fields?

Reply:
>  If so, do you have a problem with this struct existing in pubsublite/internal so classes can declare which sets of options they care about, even if at the top level those are parsed from individual google:☁️:Options fields?

I have no problem with this class existing in `pubsublite_internal`.   How they are passed via `google::cloud::Options`, it depends ..

> I think it makes it clearer to group related options,

I may not disagree that it is clearer.  That is not the only criterion.  Consistency matters.

> even if they're passed at the top level as a field in google:☁️:Options. Do you disagree?

I could read "passed at the top level as a field" two ways, it might be possible to do:

```cc
namespace g = google::cloud;
namespace psl = google::cloud::pubsublite;

void F() {
  DoStuff(g::Options{}.set<psl::BatchingOption>(psl::BatchingOptions{}
      .set_maximum_batch_message_count(100)
      .set_maximum_batch_bytes(2 * 1024 * 1024)
  ....
```

I think that is a worse experience (and less consistent with other libraries) than:

```cc
void F() {
  DoStuff(g::Options{}
      .set<psl::MaximumBatchMessageCount>(100)
      .set<psl::MaximumMatchBytes>(2 * 1024 * 1024)
 ....
```